### PR TITLE
*: add theme based frontend configuration

### PR DESCRIPTION
**Pull request by ericchiang**
This PR reworks the web layout so static files can be provided and
a "themes" directory to allow a certain degree of control over logos,
styles, etc.

This PR does NOT add general support for frontend customization,
only enough to allow us to start exploring theming internally.
The dex binary also must now be run from the root directory since
templates are no longer "compiled into" the binary.

The docker image has been updated with frontend assets.

### CODE REVIEW

**Comment by ericchiang** [comment]
cc @rithujohn191 @amrutac 

Tested the docker image with:

```
make docker-image
sudo docker run -p 5556:5556 -d \
    -v $PWD/examples/config-dev.yaml:/examples/config-dev.yaml |
    quay.io/coreos/dex:$( ./scripts/git-version ) |
    serve /examples/config-dev.yaml
./bin/example-app
```

File: server/server.go [hunk 1]
```diff
@@ -56,7 +56,32 @@ type Config struct {
...
+	// It is expected to contain the following directories:
+	//
+	//   * static - Static static served at "( issuer URL )/static".
+	//   * templates - HTML templates controlled by dex.
+	//   * themes/(theme) - Static static served at "( issuer URL )/theme".
+	//
+	Dir string
+
+	// Defaults to "( issuer URL )/theme/logo.png"
+	LogoURL string
```
Comment (amrutac) [inline server/server.go]: Just realized that I need a fav icon url as well. Any chance you can add a <code>FavIconURL</code> as well?

Comment (amrutac) [inline server/server.go]: Here's the default one.
favicon

Comment (ericchiang) [inline server/server.go]: Follow up PR?

Comment (ericchiang) [inline server/server.go]: Plus we want to either put the fav icon URL on the connector or derive it from the ID, so it'll probably take a different implementation.

**Comment by ericchiang** [comment]
per irl conversation with @amrutac we're going to punt on the fav icon. @rithujohn191 want to take a quick look?

File: server/templates.go [hunk 2]
```diff
@@ -18,74 +20,117 @@ const (
...
-			return nil, fmt.Errorf("parse files: %v", err)
+		return fmt.Errorf("stat directory %q: %v", dir, err)
+	}
+	if !stat.IsDir() {
+		return fmt.Errorf("path %q is a file not a directory", dir)
+	}
+	return nil
+}
+
+func loadWebConfig(c webConfig) (static, theme http.Handler, templates *templates, err error) {
```
Comment (rithujohn191) [inline server/templates.go]: Maybe a brief comment for this function would be helpful? Just clarifying the default behavior.

Comment (ericchiang) [inline server/templates.go]: thanks, will do

**Comment by amrutac** [comment]
👍

**Pull request merged**